### PR TITLE
[Consensus] Core to read last proposed round from DagState

### PR DESCRIPTION
**Pull request by akichidis**
## Description 

Refactoring Core to read the last proposed block from DagState rather than local cached value. This change:
* will allow for nodes that recover from amnesia (potentially) link their new proposed block the actual last proposed block instead of linking back to genesis. 
* because of the above point, we are also avoiding edge case error when an amnesia recovered node is attempting to propose for a round with exact quorum, although their last proposed block still pointing to genesis, practically making us (thankfully) hit the assert [here](https://github.com/MystenLabs/sui/blob/6a571f189b49f99ed9294cd01c82d56e03ef8bfe/consensus/core/src/core.rs#L759)
* our last proposed block is cached in DagState anyways so no practical difference

## Test plan 

CI

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:
- [ ] REST API:

### CODE REVIEW

**Review by arun-koshy — APPROVED** [review]

File: consensus/core/src/core.rs
Comment (mwtian) [inline consensus/core/src/core.rs:1505]: Blocks proposed by `Core` should have the 1st ancestor as own ancestor.

**Review by mwtian — APPROVED** [review]
I guess we will need to rebase ancestor selection on this PR?

**Comment by akichidis** [comment]
> I guess we will need to rebase ancestor selection on this PR?

Yes. If it's going to be a hassle @arun-koshy I can hold of until you merge, although the changes are quite small.

**Comment by arun-koshy** [comment]
> Yes. If it's going to be a hassle @arun-koshy I can hold of until you merge, although the changes are quite small.

Either way is fine with me! Feel free to merge and I can rebase as I am still waiting for approvals.

**Pull request merged**
